utils/compress.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python3
+from datetime import datetime
+import os
+import sys
+sys.path.append("..")
+import settings
+import re
+import tarfile
+import settings
+import boto3
+import logging
+
+S3_BUCKET_NAME = getattr(settings, "S3_BUCKET_NAME", None)
+S3_ACCESS_KEY_ID = getattr(settings, "S3_ACCESS_KEY_ID", None)
+S3_SECRET_ACCESS_KEY = getattr(settings, "S3_SECRET_ACCESS_KEY", None)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+class Compress:
+    def __init__(self, extension):
+        self.ext = extension
+        self.secret = ""
+        self.output = None
+        self.output_fn = datetime.now().isoformat()
+        self.open()
+
+    def open(self):
+        if(re.match('^\.?tgz$', self.ext, re.IGNORECASE)):
+            try:
+                self.output = tarfile.open(self.output_fn + '.tar.gz', 'w:gz')
+            except Exception as e:
+                logger.exception('[' + self.output_fn + \
+                                 '] Unable to create output file for writing ' + str(e))
+                sys.exit(0)
+        else:
+            logger.error('[' + self.output_fn + '] Invalid compression type ' + self.ext)
+            sys.exit(0)
+
+    def add(self, path):
+        if(not os.path.exists(path)):
+            logger.error('[' + self.output_fn + '] ERROR: ' + path + ' does not exist')
+            return
+        if(os.path.isfile(path)):
+            self.output.add(path, arcname=os.path.basename(path))
```

Comment:
Why is the `arcname` basename instead out `output_fn`?

Reply:
If it is a file, we want to keep the file name. If I put output_fn as arcname and it's a file, it will rename the file output_fn. 

Must provide an arcname though or else it will include the whole path of parent directories up until the file

Reply:
Hmmm ok if it's one file then it would make sense to rename it, but if I have folders and/or multiple files, I would want the tarball to contain
<datetime>.tar.gz
 |_folder1
 |_file1.png
 |_file2.png

Reply:
agree